src/codegen/js/normalize.ts

```diff
@@ -0,0 +1,115 @@
+const SANITIZE_PREFIX = 'safe_'
+
+export function sanitizeKeyword(name: string) {
+  if (!reservedWord.includes(name)) return name
+
+  return SANITIZE_PREFIX + name
+}
+
+export function sanitizeProperty(name: string) {
+  if (reservedWord.includes(name)) return SANITIZE_PREFIX + name
+  if (!reservedProperty.includes(name)) return name
+
+  return `["${name}"]`
+}
+
+const reservedProperty = [
```

Comment:
Since `sanitizeProperty` is going to be called quite frequently, I would suggest using a `Set` here and for `reservedWord`.

Reply:
You're probably right, do you have any references explaining when this kind of optimizations occurs? I don't really like black magic, Sets at least are explicit.

Reply:
Thanks for the details, I remember reading some V8 blog posts and watching videos from V8 engineers, and in a nutshell, we should write JS code as though it were statically typed (i.e. writing TypeScript is a good first step).

As I said in the design principles, perf is not my top priority as this is a toy language after all.
I was curious though and in my (not very scientific) tests, .includes is quite a bit slower with this particular length of array at least on Chrome 120 (macOS)

<img width="993" alt="image" src="https://github.com/nathsou/poy/assets/4953827/559fbef0-1b1e-4c34-844f-0dfc820267fd">
 
 https://jsperf.app/bijixof